Raven.NET.Core/Storage/RavenStorage.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Raven.NET.Core.Observers.Interfaces;
+using Raven.NET.Core.Storage.Interfaces;
+
+namespace Raven.NET.Core.Storage
+{
+    public class RavenStorage : IRavenStorage
+    {
+        public ConcurrentDictionary<Guid, string> SubjectStorage { get; } = new();
+        public ConcurrentDictionary<Type, ConcurrentDictionary<string, string>> SubjectTypeStorage { get; } = new();
+        public ConcurrentDictionary<string, IRaven> RavenWatcherStorage { get; } = new();
+        public ConcurrentDictionary<Type, IRavenTypeWatcher> RavenTypeWatcherStorage { get; } = new();
+
+        private static RavenStorage _instance;
+        private static readonly object Lock = new();
+        
+        public static RavenStorage Instance
```

Comment:
Maybe to make it little bit more flexible we can as well have nonstatic field to receive instance or even a method like GetInstance - then in IRavenStorage you can have GetInstance, and in implementation you can have it in similar way to this "Instance". If you will do this then you can as well mock this method in unit tests and In normal classes like RavenWatchers etc, you will only work on abstraction - so it can be easily moved to other cache like redis, mongo